src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,181 @@
+module Algebra.Graph.Acyclic.AdjacencyMap (
+  -- * Data types and type aliases
+  PartialOrder, 
+
+  -- * Data structure
+  AdjacencyMap, 
+
+  -- * Internal Checks
+  consistent, 
+
+  -- * Basic graph construction primitives
+  empty, vertex, overlayD, connectD, vertices,
+
+  -- * Additional functions on acyclic graphs
+  box, topSort, transitiveClosure,
+
+  -- * Acyclic graph construction methods
+  scc, fromGraph 
+  ) where
+
+import Algebra.Graph (Graph, foldg)
+import qualified Algebra.Graph.AdjacencyMap as AM
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AM
+import qualified Algebra.Graph.AdjacencyMap.Internal as AM
+import qualified Algebra.Graph.NonEmpty.AdjacencyMap as NonEmpty
+import qualified Data.Graph.Typed as Typed
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+
+{-| The 'AdjacencyMap' data type represents an acyclic graph by
+wrapping around 'Algebra.Graph.AdjacencyMap.AdjacencyMap'. The
+idea is that the user of the library should not be given a way
+to produce a cyclic graph.
+
+In the following code examples, the arthemetic is used to represent
+the graph. There is no instance of 'Num' on the AdjacencyMap so it
+used to only to act as a representation of the graph.
+-}
+newtype AdjacencyMap a = AAM
+  { aam :: AM.AdjacencyMap a
+  } deriving (Show)
+
+-- | Check if the internal graph representation is consistent,
+-- i.e. that all edges refer to existing vertices and the graph
+-- is acyclic. It should be impossible to create an inconsistent 
+-- adjacency map.
+--
+-- @
+-- consistent 'empty'         == True
+-- consistent (1 + 2)         == True
```

Comment:
Alas, we cannot implement a `Num` instance for `Acyclic.AdjacencyMap`, so we cannot use operators in the examples. All examples must be valid Haskell, and we will need to tested them as is (or with minimum possible modifications).

Reply:
See also [my comment below](https://github.com/snowleopard/alga/pull/203#issuecomment-495928972): perhaps, we could implement operators `+` and `*` that work on `AdjacencyMap Int`.

Reply:
I've mentioned at the beginning of the file that the arithmetic is only for displaying the graph structure since there no instance of 'Num'. But maybe as you said it's possible to create a safe 'Num' instance.

Reply:
I'm still exploring a type-safe version of  Num but I created a version where we filter out invalid edges.